plugin/evm/vm.go

```diff
@@ -1084,7 +1090,13 @@ func (vm *VM) initBlockBuilding() error {
 
 	if vm.ethTxGossipHandler == nil {
 		var ethTxGossipHandler p2p.Handler
-		ethTxGossipHandler, err = gossip.NewHandler[*GossipEthTx](ethTxPool, ethTxGossipHandlerConfig, vm.sdkMetrics)
+		ethTxGossipHandler = gossip.NewHandler[GossipEthTx, *GossipEthTx](
+			vm.ctx.Log,
+			vm.ethTxPushGossiper,
```

Comment:
We need to allow inbound `AppGossip` traffic from non-validators for both eth and atomic txs. Otherwise non-validators would have no way of issuing a tx.

Reply:
Should be fixed in the latest commit: https://github.com/ava-labs/coreth/pull/363/commits/d28cea07c33d5a299d13bf4e07649e500d767e56